scripts/myMangaDex.js

```diff
@@ -1237,24 +1247,65 @@ class MyMangaDex {
 		let titleInformations = {};
 
 		// collect information
+		let toUpdate = [];
+		checkTitles: // label for breaking loop
 		for (let i = 0; i < lastChapter; i++) {
 			let group = groups[i];
 			// Get title informations from LocalStorage
 			if (!(group.titleId in titleInformations)) {
 				titleInformations[group.titleId] = await storageGet(group.titleId);
-				if (titleInformations[group.titleId]) titleInformations[group.titleId].next = Infinity;
+				if (titleInformations[group.titleId]) {
+					titleInformations[group.titleId].next = Infinity;
+					titleInformations[group.titleId].getsUpdated = false;
+				}
 			}
 			// if there is data, find the next chapter
 			if (titleInformations[group.titleId]) {
 				let chapterCount = group.chapters.length;
 				for (let j = 0; j < chapterCount; j++) {
 					let chapter = group.chapters[j];
-					if (chapter.value > titleInformations[group.titleId].last && Math.floor(chapter.value) <= titleInformations[group.titleId].last + 1) {
-						titleInformations[group.titleId].next = Math.min(titleInformations[group.titleId].next, chapter.value);
+					if (chapter.value > titleInformations[group.titleId].last) {
+						if (Math.floor(chapter.value) <= titleInformations[group.titleId].last + 1) {
+							titleInformations[group.titleId].next = Math.min(titleInformations[group.titleId].next, chapter.value);
+						}
+						// if check for updates, has mal title and last checked more than 12 hours ago (12*60*60*1000ms)
+						if (checkUpdates && this.options.updateOnFollows &&
+							!titleInformations[group.titleId].getsUpdated && titleInformations[group.titleId].mal != 0 &&
+							(!titleInformations[group.titleId].lastMAL || (Date.now() - titleInformations[group.titleId].lastMAL) >= 43200000)) {
+							toUpdate.push({
+								myAnimeListId: titleInformations[group.titleId].mal,
+								mangaDexId: group.titleId,
+								lastMangaDexChapter: titleInformations[group.titleId].last,
+								chapters: titleInformations[group.titleId].chapters
+							});
+							titleInformations[group.titleId].getsUpdated = true;
+							if (toUpdate.length == 7) {
+								break checkTitles; // need to update now
+							}
```

Comment:
Could you explain how this works ? I'm not sure, I don't really like gotos and I don't understand this one, shouldn't this break go up the for loop if you set it to checkTitles ? Or does it go *after* the loop because it goes after the next goto statement ?
Also, does this update all the rows after if there is 7 to update at the beginning ?

Reply:
It's not a goto (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label)
It simply gives the outermost loop a name and break in this case doesn't break out of the current loop but out of all loops until it reaches the named on. Basically it jumps directly to after the big for loop. in other languages I would have used `break 2` to signify I want to break both loops but JS doesn't support that.

Reply:
If it has to update any, it waits for the fetches to finish then calls the function again. While fetching, no highlighting etc is done since the next chapters can change and I don't want to deal with unhiding and rehighlighting

Reply:
There should be a way to see if it's currently loading informations from MyAnimeList, if there isn't any indicator you can't know if it's broken, loading or if there is nothing to highlight.

When I wanted to implements this, I though about doing this:
1. Inspect each rows, if it needs update, put it in a temporary array and highlight it to indicate that it will be updated.
2. Continue to highlight all unrelated other chapter groups (that haven't a title that is being updated).
3. Update all chapter groups related to a title when the MyAnimeList response is received, and update highlight for it.

Reply:
Alright, I'll work on it tomorrow. What colour would you suggest for this?

Reply:
I actually had animations for this (tied to the ``is-loading`` class) and I don't know how but I can't find them in any of the CSS files.
I'll add them to your branch, you should have to use the ``.is-loading`` class only once they're here.

Reply:
I'm looking at it right now. Did you mean to add is-loading to every chapter row of the updated title? If so it looks kinda strange if multiple chapters are in a group...

Reply:
If you want, we can also talk over Discord (Consti#6699)

Reply:
On that note, I just noticed this again and I want to change it, if one manga has had the same chapter released by several groups, both chapters are marked next. Especially if there were other manga in-between this can be easily overlooked, so I want to somehow mark this. What do you think?